File: agent/src/agent_service/api/schemas/agent.py
"""Stable browser contracts for the run-scoped Agent API."""

from datetime import datetime
from typing import Any, Literal

from pydantic import BaseModel, Field


class RunSummary(BaseModel):
    id: str
    state: str
    revision: int
    provider: str
    model: str
    current_version_id: str | None
    created_at: datetime


class RunListResponse(BaseModel):
    items: list[RunSummary]
    next_cursor: str | None = None


class ResourceRecord(BaseModel):
    id: str
    created_at: datetime


class MessageRecord(ResourceRecord):
    sequence: int
    role: str
    content: str


class ProposalRecord(ResourceRecord):
    revision: int
    status: str
    affected_content: str
    suggested_replacement: str
    jd_reason: str
    source_evidence: dict[str, Any]
    evidence_request: str | None = None


class DecisionRecord(ResourceRecord):
    proposal_id: str
    proposal_revision: int
    decision: str
    version_id: str | None


class VersionRecord(ResourceRecord):
    version_number: int
    version_name: str
    parent_version_id: str | None
    content: dict[str, Any]


class ExportRecord(ResourceRecord):
    version_id: str
    content_type: str
    size_bytes: int


class ResumeSource(BaseModel):
    id: str
    filename: str
    parsed_data: dict[str, Any]


class JobDescriptionSource(BaseModel):
    id: str
    raw_text: str
    analysis: dict[str, Any]


class RunSnapshotResponse(BaseModel):
    run: RunSummary
    resume: ResumeSource | None
    job_description: JobDescriptionSource | None
    messages: list[MessageRecord]
    proposals: list[ProposalRecord]
    decisions: list[DecisionRecord]
    versions: list[VersionRecord]
    exports: list[ExportRecord]
    latest_event_sequence: int


class JobDescriptionRequest(BaseModel):
    content: str = Field(min_length=50, max_length=100_000)


class MessageRequest(BaseModel):
    content: str = Field(min_length=1, max_length=20_000)


class EvidenceRequest(BaseModel):
    content: str = Field(min_length=2, max_length=20_000)


class ProposalDecisionRequest(BaseModel):
    decision: Literal["accepted", "rejected", "revision_requested"]
    expected_revision: int = Field(ge=1)
    idempotency_key: str = Field(min_length=8, max_length=255)
    version_name: str | None = Field(default=None, max_length=255)


class RestoreVersionRequest(BaseModel):
    idempotency_key: str = Field(min_length=8, max_length=255)


class CreateExportRequest(BaseModel):
    format: Literal["pdf", "docx"]
    idempotency_key: str = Field(min_length=8, max_length=255)

File: apps/web/src/types/agent.ts
export type WorkflowState =
  | "awaiting_resume"
  | "awaiting_jd"
  | "analyzing"
  | "proposal_ready"
  | "applying"
  | "version_ready"
  | "exporting"
  | "completed"
  | "failed";

export interface RunSummary {
  id: string;
  state: WorkflowState;
  revision: number;
  provider: string;
  model: string;
  current_version_id: string | null;
  created_at: string;
}

export interface ResourceRecord { id: string; created_at: string }
export interface MessageRecord extends ResourceRecord { sequence: number; role: string; content: string }
export interface ProposalRecord extends ResourceRecord {
  revision: number; status: string; affected_content: string;
  suggested_replacement: string; jd_reason: string;
  source_evidence: Record<string, unknown>; evidence_request: string | null;
}
export interface DecisionRecord extends ResourceRecord {
  proposal_id: string; proposal_revision: number; decision: string; version_id: string | null;
}
export interface VersionRecord extends ResourceRecord {
  version_number: number; version_name: string; parent_version_id: string | null;
  content: Record<string, unknown>;
}
export interface ExportRecord extends ResourceRecord {
  version_id: string; content_type: string; size_bytes: number;
}
export interface ResumeSource { id: string; filename: string; parsed_data: Record<string, unknown> }
export interface JobDescriptionSource { id: string; raw_text: string; analysis: Record<string, unknown> }
export interface RunSnapshot {
  run: RunSummary; resume: ResumeSource | null; job_description: JobDescriptionSource | null;
  messages: MessageRecord[]; proposals: ProposalRecord[]; decisions: DecisionRecord[];
  versions: VersionRecord[]; exports: ExportRecord[]; latest_event_sequence: number;
}
export interface RunList { items: RunSummary[]; next_cursor: string | null }
export type ProposalDecision = "accepted" | "rejected" | "revision_requested";
export type ExportFormat = "pdf" | "docx";

